app/common/services/data-export.service.js

```diff
@@ -0,0 +1,104 @@
+module.exports = DataExport;
+
+
+DataExport.$inject = ['$rootScope', 'ConfigEndpoint', 'ExportJobEndpoint', '$q', '_', '$window', '$timeout', 'Notify', '$location'];
+function DataExport($rootScope, ConfigEndpoint, ExportJobEndpoint, $q, _, $window, $timeout, Notify, $location) {
+
+    function prepareExport(query) {
+        loadingStatus(true);
+        var site = ConfigEndpoint.get({ id: 'site' }).$promise;
+        var exportQuery = ExportJobEndpoint.save(query);
+
+        /**
+        TODOS:
+        1. Add polling to the api to check if the export-job is done,
+        i.e ExportJobEndpoint.get({id}) or similar, then notify the user
+        that their file is ready. 2. Keep in mind, below code will probably be changed depending on in what way the exported data is delivered
+        */
+
+        requestExport(site, query, exportQuery);
+    }
+
+    function requestExport(site, query, exportQuery) {
```

Comment:
This will be removed under the handling of downloading the data yeah?

Reply:
yep!